mlflow/data/dataset.py

```diff
@@ -0,0 +1,124 @@
+        """
+
+    @abstractmethod
+    def _to_dict(self, base_dict: Dict[str, str]) -> Dict[str, str]:
+        """
+        :param base_dict: A string dictionary of base information about the
+                          dataset, including: name, digest, source, and source
+                          type.
+        :return: A string dictionary containing the following fields: name,
+                 digest, source, source type, schema (optional), profile
+                 (optional).
+        """
+
+    def to_json(self) -> str:
+        """
+        Obtains a JSON string representation of the
+        :py:class:`Dataset <mlflow.data.dataset.Dataset>`.
+
+        :return: A JSON string representation of the
+                 :py:class:`Dataset <mlflow.data.dataset.Dataset>`.
+        """
+        base_dict = {
+            "name": self.name,
+            "digest": self.digest,
+            "source": self._source.to_json(),
+            "source_type": self._source._get_source_type(),
+        }
+        return json.dumps(self._to_dict(base_dict))
+
+    @property
+    def name(self) -> str:
+        """
+        The name of the dataset, e.g. ``"iris_data"``, ``"myschema.mycatalog.mytable@v1"``, etc.
+        """
+        if self._name is not None:
+            return self._name
+        else:
+            return "dataset"
+
+    @property
+    def digest(self) -> str:
+        """
+        A unique hash or fingerprint of the dataset, e.g. ``"498c7496"``.
+        """
+        return self._digest
+
+    @property
+    def source(self) -> DatasetSource:
+        """
+        Information about the dataset's source, represented as an instance of
+        :py:class:`DatasetSource <mlflow.data.dataset_source.DatasetSource>`. For example, this
+        may be the S3 location or the name of the managed Delta Table from which the dataset
+        was derived.
+        """
+        return self._source
+
+    @property
+    @abstractmethod
+    def profile(self) -> Optional[Any]:
+        """
+        Optional summary statistics for the dataset, such as the number of rows in a table, the
+        mean / median / std of each table column, etc.
+        """
+
+    @property
+    @abstractmethod
+    def schema(self) -> Optional[Any]:
+        """
+        Optional dataset schema, such as an instance of :py:class:`mlflow.types.Schema` representing
+        the features and targets of the dataset.
+        """
+
+    def _to_mlflow_entity(self) -> DatasetEntity:
+        """
+        :return: A DatasetEntity instance representing the dataset.
+        """
+        dataset_json = json.loads(self.to_json())
+        return DatasetEntity(
+            name=dataset_json["name"],
+            digest=dataset_json["digest"],
+            source_type=dataset_json["source_type"],
+            source=dataset_json["source"],
+            schema=dataset_json["schema"],
+            profile=dataset_json["profile"],
```

Comment:
Should we use `get` here because we say "schema (optional), profile (optional)" in `_to_dict`?